brian2/core/base.py

```diff
@@ -341,16 +331,6 @@ def brian_object_exception(message, brianobj, original_exception):
 
     See `BrianObjectException` for arguments and notes.
     '''
-    DerivedBrianObjectException = type('BrianObjectException',
-                                       (BrianObjectException, original_exception.__class__),
-                                       {})
-    new_exception = DerivedBrianObjectException(message, brianobj, original_exception)
-    # Copy over all exception attributes
-    for attribute in dir(original_exception):
-        if attribute.startswith('_'):
-            continue
-        try:
-            setattr(new_exception, attribute, getattr(original_exception, attribute))
-        except AttributeError:  # some attributes cannot be set
-            pass
-    return new_exception
+
+    raise NotImplementedError('The brian_object_exception function is no longer used. '
+                              'Raise a BrianObjectException directly.')
```

Comment:
This is just to remind ourselves not to do this in future, right?

Reply:
Yea, I wasn't sure about that. It's for the odd chance that a user (or a creator of a library) used `brian_object_exception` somewhere.